### PR TITLE
chore(upgrade): Update docs after v16 upgrade

### PR DESCRIPTION
**Pull request by MalteHerrmann**
This PR adjusts the docs with the necessary changes after the v16 upgrade. This includes updating the list of upgrades, as well as the docusaurus config.

----

Closes ENG-2405

### CODE REVIEW

**Review by 0xstepit — APPROVED** [review]

File: docusaurus.config.js [hunk 1]
```diff
@@ -59,16 +59,16 @@ const config = {
...
       rpc_url: "https://evmos.lava.build",
-      rpc_url_testnet: "https://eth.bd.evmos.dev:8545",
+      rpc_url_testnet: "https://evmos-testnet.lava.build",
       rpc_url_local: "http://localhost:8545/",
       chain_id: "9001",
       testnet_chain_id: "9000",
-      latest_version: "v13.0.2",
-      mainnet_version: "v13.0.2",
-      testnet_version: "v14.0.0-rc1",
+      latest_version: "v16.0.0",
```
Comment (ramacarlucho) [inline docusaurus.config.js:66]: Should this point to latest patch?

File: docs/validate/upgrades/list-of-upgrades.md [hunk 2]
```diff
@@ -8,40 +8,48 @@ Check the details and requirements for each mainnet and testnet upgrade.
...
-| [`v2.0.1`](https://github.com/evmos/evmos/releases/tag/v2.0.1)           |   ❌    |    ❌    |     ❌     |         ❌          | [58,700](https://www.mintscan.io/evmos/blocks/58700)        |
-| [`v2.0.0`](https://github.com/evmos/evmos/releases/tag/v2.0.0)           |   ✅    |    ✅    |     ❌     |         ❌          | [58,700](https://www.mintscan.io/evmos/blocks/58700)        |
-| [`v1.0.0`](https://github.com/evmos/evmos/releases/tag/v1.0.0) (genesis) |  `N/A`  |  `N/A`   |   `N/A`   |         ❌          | [1](https://www.mintscan.io/evmos/blocks/1)                 |
+| Version                                                                  | Planned | Breaking | Data Reset | Manual Upgrade Only | Upgrade Height                                              |
+|--------------------------------------------------------------------------|:-------:|:--------:|:----------:|:-------------------:|-------------------------------------------------------------|
+| [`v16.0.2`](https://github.com/evmos/evmos/releases/tag/v16.0.2)         |    ✅    |    ✅     |     ❌      |          ❌          | [18,295,000](https://www.mintscan.io/evmos/blocks/18295000) |
```
Comment (ramacarlucho) [inline docs/validate/upgrades/list-of-upgrades.md:13]: Shouldnt patch versions not be breaking?

**Review by ramacarlucho — APPROVED** [review]

File: docs/validate/upgrades/list-of-upgrades.md (hunk 2 above)
Comment (MalteHerrmann) [reply docs/validate/upgrades/list-of-upgrades.md:13]: yes, thanks for the catch 🙏 

```suggestion
| [`v16.0.2`](https://github.com/evmos/evmos/releases/tag/v16.0.2)         |    ✅    |    ❌     |     ❌      |          ❌          | [18,295,000](https://www.mintscan.io/evmos/blocks/18295000) |
```

File: docs/validate/upgrades/list-of-upgrades.md [hunk 3]
```diff
@@ -8,40 +8,48 @@ Check the details and requirements for each mainnet and testnet upgrade.
...
+| [`v16.0.1`](https://github.com/evmos/evmos/releases/tag/v16.0.1)         |    ✅    |    ✅     |     ❌      |          ❌          | [18,295,000](https://www.mintscan.io/evmos/blocks/18295000) |
```
Comment (MalteHerrmann) [inline docs/validate/upgrades/list-of-upgrades.md:14]:
```suggestion
| [`v16.0.1`](https://github.com/evmos/evmos/releases/tag/v16.0.1)         |    ✅    |   ❌     |     ❌      |          ❌          | [18,295,000](https://www.mintscan.io/evmos/blocks/18295000) |
```

**Pull request merged**